pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,460 @@
+
+					VersionRange: ">=1.19.0",
+
+					DependencyType: model.DependencyTypeDirect,
+
+					Justification: "test justification one",
+
+					Origin:    "Demo ingestion",
+					Collector: "Demo ingestion",
+				},
+			},
+		},
+	}
+)
+
+func ingestIsDependencyTestData(ctx context.Context, client graphql.Client) {
+
+	logger := logging.FromContext(ctx)
+
+	for _, ingest := range isDepTestData.IsDependency {
+
+		_, err := model.IsDependency(context.Background(), client, *ingest.Pkg, *ingest.DepPkg, *ingest.IsDependency)
+
+		if err != nil {
+
+			logger.Errorf("Error in ingesting: %v\n", err)
+		}
+	}
+}
+
+func Test_SearchSubgraphFromVuln(t *testing.T) {
+	server, logger := startTestServer()
+	ctx := logging.WithLogger(context.Background())
+
+	httpClient := http.Client{}
+	gqlclient := graphql.NewClient("http://localhost:9090/query", &httpClient)
+
+	ingestIsDependencyTestData(ctx, gqlclient)
+
+	var pkgIds []string
+
+	for _, dep := range isDepTestData.IsDependency {
+
+		pkgFilter := &model.PkgSpec{
+
+			Type: &dep.Pkg.Type,
+
+			Namespace: dep.Pkg.Namespace,
+			Name:      &dep.Pkg.Name,
+		}
+		pkgResponse, err := model.Packages(ctx, gqlclient, pkgFilter)
+
+		if err != nil {
+			t.Errorf("Error getting id for isDep test case: %s\n", err)
+		}
+
+		pkgIds = append(pkgIds, pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id)
+	}
+
+	testCases := []struct {
+		startID  string
+		stopID   string
+		maxDepth int
+	}{
+		{
+			// 1: test case with two dependencies at the same depth, no stopID and no limiting maxDepth
+			startID:  pkgIds[0],
+			stopID:   "",
+			maxDepth: 10,
+		},
+		{
+			// 2: test case with two levels of dependencies, no stopID and no limiting maxDepth
+			startID:  pkgIds[1],
+			stopID:   "",
+			maxDepth: 10,
+		},
+		{
+			// 3: test case with two levels of dependencies, a stopID at the first level and no limiting maxDepth
+			startID:  pkgIds[1],
+			stopID:   pkgIds[0],
+			maxDepth: 10,
+		},
+		{
+			// 4: test case with two levels of dependencies, no stopID and a limiting maxDepth at the first level
+			startID:  pkgIds[2],
+			stopID:   "",
+			maxDepth: 1,
+		},
+		{
+			// 5: test case with indirect dependency
+			startID:  pkgIds[3],
+			stopID:   "",
+			maxDepth: 1,
+		},
+		{
+			// 6: test case with isDep range that does not include the dependency
+			startID:  pkgIds[4],
+			stopID:   "",
+			maxDepth: 10,
+		},
+	}
+	for _, tt := range testCases {
+		gotMap, err := SearchDependenciesFromStartNode(ctx, gqlclient, tt.startID, tt.stopID, tt.maxDepth)
+
+		t.Run("testing searchDependenciesFromStartNode", func(t *testing.T) {
+			if err != nil {
+				t.Errorf("got err from searchDependenciesFromStartNode: %v", err)
+				return
+			}
+
+			if tt.stopID == "" && tt.maxDepth == 10 {
+				for k, v := range gotMap {
+					if !v.expanded {
+						t.Errorf("All nodes should be expanded but this node was not: node %s \n", k)
+					}
+				}
+			}
+
+			// Test Case 1
+			if tt.startID == pkgIds[0] {
```

Comment:
tests are repeated code and should probably be a loop instead over the test cases slice.

Reply:
Some of them are slightly different but I generalized the cases in order to reduce seperating out each test case / repeating code.